include/aie/AIE.td

```diff
@@ -938,11 +938,19 @@ def AIE_LockOp: AIE_Op<"lock", []>, Results<(outs Index)> {
   }];
   let arguments = (
     ins Index:$tile,
-        Confined<I32Attr, [IntMinValue<0>, IntMaxValue<15>]>:$lockID
+        OptionalAttr<Confined<I32Attr, [IntMinValue<0>, IntMaxValue<15>]>>:$lockID
+        //Confined<I32Attr, [IntMinValue<0>, IntMaxValue<15>]>:$lockID
   );
-  let assemblyFormat = [{ `(` $tile `,` $lockID `)` attr-dict }];
+  let assemblyFormat = [{ `(` $tile (`,` $lockID^ )? `)` attr-dict }];
+  //let assemblyFormat = [{ `(` $tile `,` $lockID `)` attr-dict }];
   let extraClassDeclaration = [{
-    int getLockID() { return lockID(); }
+    int getLockID() { 
+      if (lockID().hasValue()) {
+        return lockID().getValue(); 
+      } else {
+        llvm::errs() << "Lock has no ID value\n";
```

Comment:
This should give you a warning about not returning a value.  Probably this should assert, at least?